modules/trusted-firmware-m/CMakeLists.txt

```diff
@@ -271,7 +294,7 @@ if (CONFIG_BUILD_WITH_TFM)
     SOURCE_DIR ${ZEPHYR_TRUSTED_FIRMWARE_M_MODULE_DIR}
     BINARY_DIR ${TFM_BINARY_DIR}
     CONFIGURE_COMMAND ""
-    BUILD_COMMAND ${CMAKE_COMMAND} --build . ${PARALLEL_JOBS}
+    BUILD_COMMAND ${CMAKE_COMMAND} --build . --target all platform_ns ${PARALLEL_JOBS}
```

Comment:
Curious, what is the reason for explicitly building target `platform_ns` ?

Isn't this target included as part of the `all` target when `-DNS=TRUE` and target is not defined when `-DNS=FALSE` ?

Reply:
Because the TF-M build system does not depend on this target from anything in the all build, and the library has been created with `EXCLUDE_FROM_ALL`.
With NS=TRUE the tfm_api_ns library was linking with the platform_ns and the target was actually built.

Reply:
I've removed the need to platform_ns, instead the required source file is exported to the interface source files and the zephyr build system can built it, like it does for the all the other files.
This was an anomely compared to how the TF-M build system usually does it.